charts/cert-manager/v0.9.0/templates/clusterissuer.yaml

```diff
@@ -0,0 +1,88 @@
+{{- if .Values.clusterissuerEnabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "cert-manager.fullname" . }}-issuer
+  namespace: {{ .Release.Namespace }}
+data:
+  issuer.yaml: |-
+    apiVersion: certmanager.k8s.io/v1alpha1
+    kind: ClusterIssuer
+    metadata:
+      name: {{ .Values.ingressShim.defaultIssuerName }}
+      labels:
+        app: {{ template "cert-manager.name" . }}
+        chart: {{ template "cert-manager.chart" . }}
+        release: {{ .Release.Name }}
+        heritage: {{ .Release.Service }}
+    spec:
+      acme:
+        {{- if eq .Values.ingressShim.defaultIssuerName "letsencrypt-prod" }}
+        server: "https://acme-v02.api.letsencrypt.org/directory"
+        {{- else if eq .Values.ingressShim.defaultIssuerName "letsencrypt-staging" }}
+        server: "https://acme-staging-v02.api.letsencrypt.org/directory"
+        {{- else }}
+        server: {{ .Values.letsencrypt.server }}
+        {{- end }}
+        email: {{ .Values.letsencrypt.email }}
+        privateKeySecretRef:
+          name: {{ .Values.ingressShim.defaultIssuerName }}-account-key
+        http01: {}
```

Comment:
I mentioned this in [the docs PR](https://github.com/rancher/docs/pull/1691#issuecomment-526662805), and it's even more relevant here.

Basically, there's a [new solver format in 0.8](https://docs.cert-manager.io/en/latest/tasks/upgrading/upgrading-0.7-0.8.html) that deprecates the old way of doing it. The old way still works, but it'll be removed before 1.0, and they're about to roll out with 1.0-alpha. Users who don't upgrade the ClusterIssuer will be surprised.

It's a bit chicken/egg. `ingress-shim` certs will only use the new solver if the old config is removed. Prematurely removing the old config can break certs not issued by `ingress-shim`. So perhaps we can check if they have non-`ingress-shim` certs, and if not, make the update to the ClusterIssuer for them? 

What happens if someone installs our app, manually updates the ClusterIssuer, and then upgrades from the app to a newer version? Will their solver update be overwritten?

There was also [an issue with the upgrade to 0.9](https://github.com/jetstack/cert-manager/issues/1927) that required `--force` to recreate the deployments because they couldn't be patched with new ClusterRoles and ClusterRoleBindings being installed at the same time. Is our app immune to that?

Finally, in my work to create [the video for this](https://youtu.be/nF_k1YZYjH0), I found that certs generated by `ingress-shim` wouldn't update until the secret was removed. Then they updated and issued a new secret. They _might_ update automatically at the next renewal.